include/circt/Dialect/FIRRTL/FIRRTLInstanceGraph.h

```diff
@@ -14,32 +14,32 @@
 #define CIRCT_DIALECT_FIRRTL_FIRRTLINSTANCEGRAPH_H
 
 #include "circt/Dialect/FIRRTL/FIRRTLOps.h"
-#include "circt/Dialect/HW/InstanceGraphBase.h"
+#include "circt/Support/InstanceGraph.h"
 #include "circt/Support/LLVM.h"
 #include "llvm/ADT/GraphTraits.h"
 #include "llvm/ADT/STLExtras.h"
 #include "llvm/ADT/iterator.h"
 
 namespace circt {
 namespace firrtl {
-using InstanceRecord = hw::InstanceRecord;
-using InstanceGraphNode = hw::InstanceGraphNode;
-using InstancePathCache = hw::InstancePathCache;
+using InstanceRecord = igraph::InstanceRecord;
+using InstanceGraphNode = igraph::InstanceGraphNode;
+using InstancePathCache = igraph::InstancePathCache;
 
 /// This graph tracks modules and where they are instantiated. This is intended
 /// to be used as a cached analysis on FIRRTL circuits.  This class can be used
 /// to walk the modules efficiently in a bottom-up or top-down order.
 ///
 /// To use this class, retrieve a cached copy from the analysis manager:
 ///   auto &instanceGraph = getAnalysis<InstanceGraph>(getOperation());
-class InstanceGraph : public hw::InstanceGraphBase {
```

Comment:
I'm not completely convinced we need per-dialect versions of the InstanceGraph, but that isn't blocking for this.